presto-spi/src/main/java/com/facebook/presto/spi/resourceGroups/ResourceGroupSelectionContext.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.resourceGroups;
+
+import java.util.Objects;
+import java.util.OptionalInt;
+
+import static java.util.Objects.requireNonNull;
+
+public class ResourceGroupSelectionContext
+{
+    private final OptionalInt dynamicSegmentPosition;
+
+    private ResourceGroupSelectionContext(OptionalInt dynamicSegmentPosition)
+    {
+        this.dynamicSegmentPosition = requireNonNull(dynamicSegmentPosition, "dynamicSegmentPosition is null");
+    }
+
+    public static ResourceGroupSelectionContext createResourceGroupSelectionContext(int dynamicSegmentPosition)
+    {
+        return new ResourceGroupSelectionContext(OptionalInt.of(dynamicSegmentPosition));
+    }
+
+    public static ResourceGroupSelectionContext createResourceGroupSelectionContext(OptionalInt dynamicSegmentPosition)
+    {
+        return new ResourceGroupSelectionContext(dynamicSegmentPosition);
+    }
```

Comment:
do we really need both of these? Should we just keep the `OptionalInt` one?

Reply:
I will keep the OptionalInt.